library/core/src/fmt/rt.rs

```diff
@@ -53,3 +51,147 @@ pub enum Count {
     /// Not specified
     Implied,
 }
+
+// This needs to match the order of flags in compiler/rustc_ast_lowering/src/format.rs.
+#[derive(Copy, Clone)]
+pub enum Flag {
```

Comment:
Does it make sense to use `pub(super)` here so it's clear it's not used outside core? Ditto the rest of these moved types.

Reply:
Yes. But only for this one enum, because all the other types here are lang items that need to be `pub` so they can be used in the hir expansion of `format_args!()`.

Updated.